### DISCUSSION TITLE
defaultValue not being set correctly..

### DISCUSSION THREAD

**Discussion by svayam-support** [sequelize/sequelize#17003] [Help & Questions]
I am using sequelize version 6.34 with mariadb 10.5. 

(a) If I use following attribute definition to define my model and sync, I am getting a function with string and not a function with column identifier in the generated default function.
attribute definition used - si.define(act_id+"_di", 
      {
        // Model attributes are defined here
        di_id           : { type: DataTypes.INTEGER({unsigned:true}), allowNull: false, primaryKey: true, autoIncrement: true, },
        di_usr_unq_cd   : { 
          type: DataTypes.CHAR(50), 
          allowNull: true,
          defaultValue : Sequelize.fn('sqrt',[`di_id`]),
          ....
table created -
CREATE TABLE `di` (
  `di_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `di_usr_unq_cd` char(50) DEFAULT sqrt('di_id'),   <--  this should be column identifier with ticks like `di_tcd`
    ....

(b) If I switch to using Sequelize.col instead, I get error when syncing ...
Alternate definition used - 
{
        // Model attributes are defined here
        di_id           : { type: DataTypes.INTEGER({unsigned:true}), allowNull: false, primaryKey: true, autoIncrement: true, },
        di_usr_unq_cd   : { 
          type: DataTypes.CHAR(50), 
          allowNull: true,
          defaultValue : Sequelize.fn('sqrt',[Sequelize.col('di_id')]),
          ....
Error trace -
uncaughtException: Invalid value Col { col: 'di_id' }
Error: Invalid value Col { col: 'di_id' }
    at escape (/Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/sql-string.js:53:11)
    at /Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/sql-string.js:12:14
    at Array.reduce (<anonymous>)
    at arrayToList (/Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/sql-string.js:5:16)
    at Object.escape (/Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/sql-string.js:50:12)
    at MariaDBQueryGenerator.escape (/Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/dialects/abstract/query-generator.js:763:22)
    at /Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/dialects/abstract/query-generator.js:1699:21
    at Array.map (<anonymous>)
    at MariaDBQueryGenerator.handleSequelizeMethod (/Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/dialects/abstract/query-generator.js:1692:38)
    at MariaDBQueryGenerator.handleSequelizeMethod (/Users/rkant/bkup/github/4_svayam/902-DD/7-libraries/svm-test-ts/node_modules/sequelize/lib/dialects/mysql/query-generator.js:252:18)

**Comment by ephys** [comment]
You simply need to remove the array:

```
fn('sqrt', col('di_id'))
```

**Reply by svayam-support** [reply]
Thanks ephys. What is I have more than one parameters like Sequelize.fn('concat',[Sequelize.col('di_id'), Sequelize.col('di_tcd')])

**Reply by svayam-support** [reply]
Also, I tried the solution you suggested and was able to successfully get the right default condition into the table -

# attribute definition used - si.define(act_id+"_di",
{
// Model attributes are defined here
di_id : { type: DataTypes.INTEGER({unsigned:true}), allowNull: false, primaryKey: true, autoIncrement: true, },
di_usr_unq_cd   : { 
          type: DataTypes.CHAR(50), 
          allowNull: false,
          defaultValue: si.fn('concat',si.col('di_tcd'),"_",si.col('di_tech_nm'))
        },
....
# table created -
CREATE TABLE `di` (
  `di_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `di_usr_unq_cd` char(50) NOT NULL DEFAULT concat(`di_tcd`,"_",`di_tech_nm`), <-- ticks are correct now.

# But, at the time of row insertion, every row is getting defaulted to only "_" without the values of columns `di_tcd` and `di_tech_nm`. Is there a way to stop sequelize from trying to set the default value when the function has been passed on to the database for calculating default column value.

**Reply by ephys** [reply]
Could you edit your post to use code blocks? It's very difficult to read what the code is

That being said, for default values like `fn`, Sequelize does not insert any value itself, it delegates to the database

If it's not working, try running the query manually to see if it's a limitation of the database. I'd test it myself but I'm not on a dev computer right now
